### PR TITLE
Ca develop

### PR DESCRIPTION
**Pull request by lisa-bengtsson**
Cellular automata updates to be able to run global and sub-grid CA's at the same time, improvements for stochastic convection representation, some cleaning. Updates are dependent upon updates in submodules FV3 and ccpp/physics.

### CODE REVIEW

**Comment by lisa-bengtsson** [comment]
> I code changes look good. Just to confirm, this will also require an update to fv3_atm? If so, I will hold off doing the merge until that gets coordinated with Jun.

That is correct. Let me know if I can help with that coordination somehow.

**Comment by climbfuji** [comment]
I can take care of that. We currently have it scheduled for April 14.

**Comment by climbfuji** [comment]
Associated PRs:

https://github.com/NCAR/ccpp-physics/pull/418
https://github.com/noaa-psd/stochastic_physics/pull/15
https://github.com/NOAA-EMC/fv3atm/pull/87
https://github.com/ufs-community/ufs-weather-model/pull/96

File: cellular_automata_sgs.F90 [hunk 1]
```diff
@@ -42,93 +46,73 @@ subroutine cellular_automata(kstep,Statein,Coupling,Diag,nblks,nlev, &
...
 
-k350 = 29
-k850 = 13
-alpha = 1.5
-ra = 201
-rb = 2147483648
-rc = 4294967296
-
 halo=1
 k_in=1
```
Comment (pjpegion) [inline cellular_automata_sgs.F90:82]: I suggest a code block such as:
if (nlev .EQ. 64) then
   k350=29
   k850=13
else (nlev. EQ. 127) then
  k350=61
  k850=28
else ! make a guess
   k350=nint(nlev/2)
   k850=nint(nlev/5)
   print*,'this level selection is not supported, making an approximation for k350 and k850'
endif

Comment (lisa-bengtsson) [reply cellular_automata_sgs.F90:82]: Thanks, good suggestion!

Comment (lisa-bengtsson) [reply cellular_automata_sgs.F90:82]: I addressed this and results are reproduced.

Comment (pjpegion) [reply cellular_automata_sgs.F90:82]: good,  once you commit and push the updates, I can do the merge.

**Pull request merged**